### PR TITLE
add toString method

### PR DESCRIPTION
**Pull request by p4veI**
* add a standard public toString method alongside the magic __toString methods

motivation:
* helps to avoid explicit type casts over Token/Plain class in v4

### CODE REVIEW

File: src/Token/DataSet.php [hunk 1]
```diff
@@ -49,8 +49,13 @@ public function all(): array
...
-    public function __toString(): string
+    public function toString(): string
     {
         return $this->encoded;
     }
+
+    public function __toString(): string
+    {
+        return $this->toString();
+    }
```
Comment (lcobucci) [inline src/Token/DataSet.php:60]: It think it feels pointless to have this here if we won't use it internally.

File: src/Token/Signature.php [hunk 2]
```diff
@@ -34,8 +34,13 @@ public function hash(): string
...
-    public function __toString(): string
+    public function toString(): string
     {
         return $this->encoded;
     }
+
+    public function __toString(): string
+    {
+        return $this->toString();
+    }
```
Comment (lcobucci) [inline src/Token/Signature.php:45]: Same remark, we can just get rid of this method

File: src/Token.php [hunk 3]
```diff
@@ -51,5 +51,10 @@ public function isExpired(DateTimeInterface $now): bool;
     /**
      * Returns an encoded representation of the token
      */
+    public function toString(): string;
+
+    /**
+     * Returns an encoded representation of the string while casting Token to string
+     */
     public function __toString(): string;
```
Comment (lcobucci) [inline src/Token.php:59]: Let's really drop it.
On v3.4 we mark it as `@deprecated` and introduce `toString()` as well.

Comment (p4veI) [reply src/Token.php:59]: was worried it wouldn't be approved if I drop it :) cheers, will update now.

Comment (lcobucci) [reply src/Token.php]: Thanks, it was a bad decision back then. I just had bigger fishes to fry :sweat_smile:

**Pull request merged**